rust/src/pgsql/pgsql.rs

```diff
@@ -497,7 +533,9 @@ impl PgsqlState {
                         } else {
                             tx.responses.push(response);
                             if tx_completed {
-                                tx.tx_state = PgsqlTransactionState::ResponseDone;
+                                tx.tx_req_state = PgsqlTxReqProgress::TxReqDone;
```

Comment:
Feels strange that a response updates the request state...

Reply:
The reasoning was that per the protocol, there shouldn't be any other requests associated with this transaction if the transaction is done in the to_client direction.